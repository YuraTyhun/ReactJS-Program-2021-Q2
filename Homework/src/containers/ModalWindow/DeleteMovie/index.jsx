import React from 'react';

import { useSelector } from 'react-redux';

import Button from '../../../components/Button';

const DeleteMovie = ({ title, movieId }) => {
    const activeModal = useSelector(({movie: {activeModal}}) => activeModal);

    return activeModal === 'delete' && (
        <>
            <h1 className="modal-window-title">{title}</h1>
            <h3 className="modal-window-content">Are you sure you want to delete this movie?</h3>
            <div className="modal-window-action">
                <Button
                    className="modal-window-action-btn"
                    title="CONFIRM"
                />
            </div>
        </>
    )
};

export default DeleteMovie;