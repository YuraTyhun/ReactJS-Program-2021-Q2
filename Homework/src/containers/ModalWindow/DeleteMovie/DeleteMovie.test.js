import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import DeleteMovie from './index';

it('renders deleteMovie form correctly', () => {
    const mockStore = configureMockStore();
    const initState = {
        movie: {
            activeModal: 'delete'
        }
    };
    
    const store = mockStore(initState);
    const deleteMovie = renderer.create(
        <Provider store={store}>
            <DeleteMovie title="DELETE MOVIE" movieId="5"/>
        </Provider>
    ).toJSON();

    expect(deleteMovie).toMatchSnapshot();

});