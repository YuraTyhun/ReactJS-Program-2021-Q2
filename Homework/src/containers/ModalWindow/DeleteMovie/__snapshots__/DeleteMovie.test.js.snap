// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders deleteMovie form correctly 1`] = `
Array [
  <h1
    className="modal-window-title"
  >
    DELETE MOVIE
  </h1>,
  <h3
    className="modal-window-content"
  >
    Are you sure you want to delete this movie?
  </h3>,
  <div
    className="modal-window-action"
  >
    <button
      className="modal-window-action-btn"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      CONFIRM
      
    </button>
  </div>,
]
`;
